pkg/apis/workflow/v1alpha1/workflow_types.go

```diff
@@ -2362,14 +2397,38 @@ type DAGTask struct {
 	// OnExit is a template reference which is invoked at the end of the
 	// template, irrespective of the success, failure, or error of the
 	// primary template.
+	// DEPRECATED: Use LifecycleHook.template instead.
 	OnExit string `json:"onExit,omitempty" protobuf:"bytes,11,opt,name=onExit"`
 
 	// Depends are name of other targets which this depends on
 	Depends string `json:"depends,omitempty" protobuf:"bytes,12,opt,name=depends"`
+
+	// Hooks hold exit hook which is invoked at the end of the
+	// workflow, irrespective of the success, failure, or error status of the primary template
```

Comment:
```
Hooks are invoked during the lifecycle of the template.
```
I don't think you should comment on exit hooks here. It duplicates other comments.